kotlin/display-content-of-utility-network-container/src/main/java/com/esri/arcgisruntime/sample/displaycontentofutilitynetworkcontainer/MainActivity.kt

```diff
@@ -0,0 +1,393 @@
+        exitButton.setOnClickListener {
+            handleExitButtonClick()
+        }
+        // display the legend alert dialog when the button is clicked
+        legendButton.setOnClickListener {
+            legendDialog?.show()
+        }
+    }
+
+    /**
+     * Called when the user exits the container view,
+     * clears graphics, resets the viewpoint
+     * and displays the operationalLayers
+     */
+    private fun handleExitButtonClick() {
+        graphicsOverlay.graphics.clear()
+        mapView.setViewpointAsync(previousViewpoint)
+        mapView.map.operationalLayers.forEach { layer ->
+            layer.isVisible = true
+        }
+        // exits from the container view
+        handleContainerView(false)
+    }
+
+    /**
+     * Show legend, exit button and disable interaction
+     * when container view [isVisible]
+     */
+    private fun handleContainerView(isVisible: Boolean) {
+        if (isVisible) {
+            // enable buttons
+            exitButton.isEnabled = true
+            legendButton.isEnabled = true
+            // disable map interactions
+            mapView.interactionOptions.apply {
+                isPanEnabled = false
+                isZoomEnabled = false
+            }
+        } else {
+            // disable button
+            exitButton.isEnabled = false
+            legendButton.isEnabled = false
+            // enable map interactions
+            mapView.interactionOptions.apply {
+                isPanEnabled = true
+                isZoomEnabled = true
+            }
+        }
+    }
+
+    /**
+     * Set up the attachmentSymbol, connectivitySymbol and
+     * the boundingBoxSymbol bitmaps for the legend view
+     */
+    private fun setUpLegendView() {
+        // inflate the layout and get references to each of its components
+        val dialogBinding = UtilityNetworkLegendBinding.inflate(LayoutInflater.from(this))
+        // set the image bitmaps using the line symbols to the ImageViews
+        dialogBinding.apply {
+            attachmentImageView.setImageBitmap(
+                attachmentSymbol.createSwatchAsync(0x00000000, 1F).get()
+            )
+            connectivityImageView.setImageBitmap(
+                connectivitySymbol.createSwatchAsync(0x00000000, 1F).get()
+            )
+            boundingImageView.setImageBitmap(
+                boundingBoxSymbol.createSwatchAsync(0x00000000, 1F).get()
+            )
+        }
+
+        // create the alert dialog and bind it to the inflated layout
+        legendDialog = AlertDialog.Builder(this).apply {
+            setView(dialogBinding.root)
+            setTitle("Legend")
+        }.create()
+    }
+
+    /**
+     * Identifies the feature of the clicked [mapPoint], gets the selected feature's sublayerResults
+     * and retrieves the utility network associations of the selected ArcGISFeature
+     */
+    private fun handleMapViewClicked(mapPoint: android.graphics.Point) {
```

Comment:
I am noticing some unexpected behaviour: If you click on the same feature multiple times, it creates multiple bounding boxes, and if you click on two features in quick succession it creates one bounding box that covers both features. I think it could be fixed by showing the progress bar before starting the identify call (I'm assuming the progress bar blocks motion events from getting to the MapView?).

Reply:
That is definitely unexpected behavior. Showing the progress bar right after the map click in this case does not block motion events from getting to the MapView. Instead added an if check to see if the `progressBar` is loading

```kotlin
// handle map click only if progressBar is not loading
if(progressBar.visibility == View.GONE){
    // display the progressBar
    progressBar.visibility = View.VISIBLE
    ....
    handleMapViewClicked(screenPoint)
}
```